Added embeds for other commands

Fountai2 opened the pull request:
Abstracted most of the embed logic into the the constructor, can probably do more later

Did some formatting based on when the game was.

Added a limit of 20 to use the embed, otherwise it will use list

Putting up for now. Will add Degen team formatting later.

Fountai2 committed:
Added embeds for other commands
Abstracted most of the embed logic into the the constructor, can probably do more later

Did some formatting based on when the game was.

Added a limit of 20 to use the embed, otherwise it will use list

jschultz38 commented on utils/DegenEmbed.py:
Since these are just generator methods, I think this should just be a util method

Fountai2 replied:
What does this mean?

Fountai2 replied:
The create method allows some more variability in the images we add to it.

e.g. the memes - I set the thumbnail in the meme creation - that's static and won't change for the given command. 

For the game specific commands having the create function in the class allows us to dynamically change the displayed image contextually. In the current use case I do a look up on creation to determine what image to use based on the degen playing. 

The other benefit to putting it in the class allows for future flexibility wrt to overriding the embed. 

Also for the game logic I really wanted to make the bot script simpler and only have one line for creating the embed.

jschultz38 commented on utils/DegenEmbed.py:
```diff
@@ -1,25 +1,36 @@
+
+        for game in games:
+            if game.gametime <= datetime.now():
+                newembed.add_field(f'{game.away_team}: {game.away_score} '
```
Stick with the format that we've been using for games, "^{degen_team}: 2, {opponent_team}: 0"

Fountai2 replied:
I'll double check this but I don't think we make this super easy. We should probably just transform the data object to have degen_team and shit_team or some easier way to figure it out.

jschultz38 commented on utils/DegenEmbed.py:
```diff
@@ -1,25 +1,36 @@
+            if game.gametime <= datetime.now():
+                newembed.add_field(f'{game.away_team}: {game.away_score} '
+                                   f'{game.home_team}: {game.home_score}',
+                                   f'{game.gametime.strftime('%B %d')}',
```
I don't see why this gametime should be any different than an upcoming game gametime

Fountai2 replied:
I did this because it was too busy for information no one really cares about. If the game happened, the time it happened isn't super relevant. People might care about the score or when the game was but beyond that the timestamp was busy. 

![image](https://github.com/jschultz38/degenbot/assets/2738715/b30d2b93-7c03-4e12-a313-c00c4a978474)
 Looked better than 
 
![image](https://github.com/jschultz38/degenbot/assets/2738715/5a016822-0e6c-425f-9b43-b6e7ba0c3400)

It also created a visual difference so it was easier to parse out what was past.

jschultz38 commented on utils/DegenEmbed.py:
```diff
@@ -1,25 +1,36 @@
+                                   inline=False)
+            else:
+                newembed.add_field(f'{game.away_team} Vs. {game.home_team} ',
+                                   f'{game.gametime} @ {game.location}', inline=False)
```
I like this format, but the gametime needs a strftime to make it look better

Fountai2 replied:
Yeah I agree but I feel like that should be done upstream and not here.

Edit to add: doing it here means that where we don't use the embed formatting we have to duplicate code.

Fountai2 replied:
Coffee thought this morning we could replace the timestamps with links to the VOD if the game is past - since people seem to be using this bot as a way to reference their own schedules.

jschultz38 replied:
yeah i think we should add a link to the web page as well

Fountai2 committed:
confirsted construct full to static

removed the old create method

Added conditional logic for past games to declare victory and defeat and just make it prettier

did embed level string formatting for times

renamed game to HockeyGame for consistency.

made degen_home a bool

Added a game level victory status but that should be set from the page content to remove redundancy later.

Changed how we call out degen teams - using markup to bold it.

jschultz38 committed:
Updates and fixed for DegenEmbed

1. Remove DegenEmbed as a class and just make generator methods
2. minor fix for !next embed
3. Add accompanying fixes for HockeyGame bool change

Signed-off-by: Jarrett Schultz <jschultz38@gatech.edu>

jschultz38 committed:
Add a show players option for game embed gen

When we use !today, we should show the players that will be
participating in each game

Signed-off-by: Jarrett Schultz <jschultz38@gatech.edu>
